### DISCUSSION TITLE
Add support for GVfs in FeatherPad

### DISCUSSION THREAD

**Discussion by kozak243** [tsujan/FeatherPad#771] [Ideas]
I'm a big fan of Featherpad and use it regularly for text editing. However, there are certain directories that I don't have direct permissions to access and I'd love to be able to edit files located there through Featherpad instead of having to use the command line or chmod/chown them. Adding support for an admin backend would make my life much easier and workflow smoother.
(Also, I would change 'The file doesn't exist' to 'The file doesn't exist or is inaccessible' or something along those lines.)
I have multiple systems on my PC and accessing them often requires admin privileges. I don't want to change permissions of things there as it may cause issues in the future.

**Comment by tsujan** [comment]
Not being based on GLib, FeatherPad can't use GVFS' admin functionality — in case you meant it.

Of course, FeatherPad already supports saving as root through Polkit where there is a read permission (you need a Polkit authentication agent, which is provided by most DEs).

It can also be launched as root (even under Wayland and with a well-known method) through a *safe graphical front-end* for `sudo` (`sudo` itself should **never** be used), although that's not recommended.

**Reply by kozak243** [reply]
Hi! Thanks for the response

I didn't know software have to be build with GLib to support admin protocol. I believed it was just a matter of sending some requests or/and forwarding some info to GVfs and it handled the rest. I thought it works like an API. My bad, sorry.

**Reply by tsujan** [reply]
No need to be sorry; perhaps I should have explained it more: GVFS works with GIO (Gnome Input/Output), which belongs to GLib. All apps based on GLib support it automatically, without needing it as a build dependency — that's a nice thing about GVFS.

GLib is made by GNOME team but isn't limited to GNOME. And being written in Qt doesn't mean an app can't use GLib. For example, pcmanfm-qt is written in Qt but uses GLib for file management and supports GVFS's admin mode (among other things). However, FeatherPad uses purely Qt methods.
